### ISSUE TITLE
RCS1146 should not be reported in expressions

### ISSUE THREAD

**Issue by nvmkpk** [JosefPihrt/Roslynator#792]
**Product and Version Used**: 
VS Extension 3.1.0

**Steps to Reproduce**:
Use one of the code below:
```csharp
Expression<Func<int?, string>> expression = x => x?.ToString();
```
```csharp
IQueryable<T> query = from x in myDbContext.MyEntities
                      where x.Abc?.IsTrue
                      select x.Xyz;
```

**Actual Behavior**:
After applying the code fix compiler reports CS8072 as null propagation operator is not allowed in expressions.

**Expected Behavior**:
No RCS1146 is reported in expression.

**Comment by JosefPihrt** [comment]
Your code does not produce any RCS1146 diagnostic. It seems that it is actually the code after the code fix was applied. Could you provide the code that produces RCS1146?
